src/main/java/gov/cabinetofice/gapuserservice/service/user/OneLoginUserService.java

```diff
@@ -184,4 +194,20 @@ public void invalidateUserJwt(final Cookie customJWTCookie, final HttpServletRes
         thirdPartyAuthToken.setMaxAge(0);
         response.addCookie(thirdPartyAuthToken);
     }
+
+    public void validateRoles(List<Role> userRoles, String payloadRoles) throws UnauthorizedException {
+        final List<String> formattedPayloadRoles = removeSquareBracketsAndTrim(Arrays.asList(payloadRoles.split(",")));
+        final Set<String> formattedUserRoles = userRoles.stream()
+                .map(role -> roleMapper.roleToRoleDto(role).getName())
+                .collect(Collectors.toSet());
+
+        if(!formattedUserRoles.containsAll(formattedPayloadRoles)){
+            throw new UnauthorizedException("Roles in payload do not match roles in database");
+        }
+    }
+
+    public void validateAdminSession(String emailAddress, String roles) throws UnauthorizedException {
```

Comment:
Any reason you have throws UnauthorizedException in the method signature? i think you can remove this from here and every method that calls it

Reply:
Leftover from when `validateRoles` above wasn't extracted. Have removed

Reply:
Doesn't look like it's been removed?

The method above also doesn't need to have it in the method signature

![Screenshot 2023-09-13 at 14 48 40](https://github.com/cabinetoffice/gap-user-service/assets/135221918/1d836596-8f6d-4cee-b8e6-c8ecb85661cc)

Reply:
Fixed